### PR TITLE
Improve behavior of std.socket "soft" unittests

### PR DESCRIPTION
**Pull request by CyberShadow**
See discussion in https://github.com/dlang/phobos/pull/7579 for details.

### CODE REVIEW

**Comment by WalterBright** [comment]
Please change commit message to:

"fix Issue 21114"

**Comment by CyberShadow** [comment]
I don't know why @dlang-bot thinks the issue reference is not fixing it. I copied its code to a small program and it correctly detects that `closed` is `true`.

File: std/socket.d [hunk 1]
```diff
@@ -1709,18 +1691,18 @@ public:
...
+        const InternetAddress ia = new InternetAddress("255.255.255.255", 80);
         assert(ia.toHostNameString() is null);
 
         if (getnameinfoPointer)
         {
             // test failing reverse lookup, via gethostbyaddr
             auto getnameinfoPointerBackup = getnameinfoPointer;
-            getnameinfoPointer = null;
-            scope(exit) getnameinfoPointer = getnameinfoPointerBackup;
+            cast()getnameinfoPointer = null;
```
Comment (wilzbach) [inline std/socket.d:1704]:
```suggestion
            cast() getnameinfoPointer = null;
```

File: std/socket.d [hunk 2]
```diff
@@ -1709,18 +1691,18 @@ public:
...
+            scope(exit) cast()getnameinfoPointer = getnameinfoPointerBackup;
```
Comment (wilzbach) [inline std/socket.d:1705]:
```suggestion
            scope(exit) cast() getnameinfoPointer = getnameinfoPointerBackup;
```

**Comment by wilzbach** [comment]
> I don't know why @dlang-bot thinks the issue reference is not fixing it.

Not sure either, my best guess is that the bot is confused with the `std.socket` prefix, see e.g. this test:

https://github.com/dlang/dlang-bot/blob/master/source/dlangbot/bugzilla.d#L61

**Comment by CyberShadow** [comment]
> I copied its code to a small program and it correctly detects that `closed` is `true`.

Ah, whoops, I must have been looking at the wrong branch. I can reproduce with the code from the current master you linked.

**Comment by wilzbach** [comment]
The RegExp was changed recently to cater more for the GH-syntax people are used to, see https://github.com/dlang/dlang-bot/commit/b7efd0b71355033d50f371afda3fde62cbcc6dbc

**Comment by CyberShadow** [comment]
> The RegExp was changed recently to cater more for the GH-syntax people are used to, see [dlang/dlang-bot@b7efd0b](https://github.com/dlang/dlang-bot/commit/b7efd0b71355033d50f371afda3fde62cbcc6dbc)

Thanks, here's the fix: https://github.com/dlang/dlang-bot/pull/254

**Pull request merged**